tests/system/src/test/java/test/robot/javafx/scene/treetableview/TreeTableViewChangeRootTest.java

```diff
@@ -0,0 +1,160 @@
+
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) x, (int) y);
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+
+        Util.sleep(500);
+
+        Util.runAndWait(() -> {
+
+            Node disclosure = scene.lookup(".tree-disclosure-node");
+            if (disclosure == null) {
+                throw new IllegalStateException("Disclosure node not found.");
+            }
+
+            Bounds dbounds = disclosure.localToScreen(disclosure.getBoundsInLocal());
+            double dx = dbounds.getMinX() + dbounds.getWidth() / 2;
+            double dy = dbounds.getMinY() + dbounds.getHeight() / 2;
+
+            robot.mouseMove((int) dx, (int) dy);
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+
+        Util.sleep(1000);
+
+        Set<Node> renderedRows = tree.lookupAll(".tree-table-row-cell");
+        List<String> rendered = renderedRows.stream()
+                .filter(n -> n instanceof TreeTableRow<?>)
+                .map(n -> ((TreeTableRow<?>) n).getItem())
+                .filter(Objects::nonNull)
+                .map(Object::toString)
+                .toList();
+
+        assertTrue(rendered.containsAll(List.of("V0", "V1", "V2", "V3", "V4")),
+                "Rows should contain 5 children after expanding the root");
+    }
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+            tree = new TreeTableView<>();
+            tree.setShowRoot(true);
+
+            TreeTableColumn<String, String> column = new TreeTableColumn<>("Name");
+            column.setCellValueFactory(param -> new SimpleStringProperty(param.getValue().getValue()));
+            column.setPrefWidth(300);
+            tree.getColumns().add(column);
+
+            changeRootButton = new Button("Change Root");
+            changeRootButton.setOnAction(e -> {
+                TreeItem<String> newRoot = new TreeItem<>();
+                for (int i = 0; i < 5; i++) {
+                    newRoot.getChildren().add(new TreeItem<>("V" + i));
+                }
+                tree.setRoot(newRoot);
+            });
+
+            VBox layout = new VBox(10, changeRootButton, tree);
+            scene = new Scene(layout, SCENE_WIDTH, SCENE_HEIGHT);
+            stage.setScene(scene);
+            stage.initStyle(StageStyle.UNDECORATED);
+            stage.setAlwaysOnTop(true);
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> Platform.runLater(startupLatch::countDown));
+            stage.show();
+
+            // Pre-populate with 1 item to start
+            TreeItem<String> initialRoot = new TreeItem<>();
+            initialRoot.getChildren().add(new TreeItem<>("V0"));
+            tree.setRoot(initialRoot);
+        }
+    }
+}
```

Comment:
missing newline

(it's always a good idea to go through the files once the PR is created)